lib/active_record_extended/query_methods/where_chain.rb

```diff
@@ -85,6 +93,23 @@ def left_column(arel)
       @scope.klass.columns_hash[arel.left.name] || @scope.klass.columns_hash[arel.left.relation.name]
     end
 
+    def equality_to_key_check(kind, opts, rest)
+      build_where_chain(opts, rest) do |arel|
+        case arel
+        when Arel::Nodes::In, Arel::Nodes::Equality, Arel::Nodes::HomogeneousIn
+          column = left_column(arel) || column_from_association(arel)
+
+          if [:hstore, :jsonb].include?(column.type)
+            kind.new(arel.left, arel.right)
+          else
+            raise ArgumentError.new("Invalid argument for .where.#{kind}(), got #{arel.class}")
```

Comment:
This is not user friendly (`kind` is a ruby class) but just leaving it here.
If the feature makes sense some cleanup and documentation should be added as well :)